### ISSUE TITLE
Unable to detect isotopic peaks if unlabelled peak is not present

### ISSUE THREAD

**Issue by sunil20dhakad** [ElucidataInc/ElMaven#687]
I am working with labelled untargeted LCMS data. I want to detect all possible isotopes for the metabolites but currently El-Maven is not able to pick those peaks whose unlabelled form is not present i.e. it ignores the isotopic peaks. For eg. in case of my data there is no unlabeled peak for dCTP but its isotopic peaks are present which El-Maven is not able to pick. 
@shefalilathwal @sahil21

**Comment by chubukov** [comment]
I guess my judgment was off :) #22

**Comment by chubukov** [comment]
So I am not sure if something has changed recently, or if my understanding of this bug was incorrect. I previously though this came up only if there was no parent peak in any sample, which would be sufficiently unusual. 

But working through some recent data, I see now that you lose all labeled peaks for _any_ _sample_  where there is no parent peak, even if it's present in other samples. I.e. if you get 100% labeling, you'll lose the peak entirely. That seems to me like a huge bug.

**Comment by shubhra-agrawal** [comment]
@chubukov That's correct. El-MAVEN needs a parent peak in each sample to get its corresponding isotopic peak. This is because the algorithm looks for possible isotopic m/zs within a 5 scan range of the parent peak scan (the peak scan number differs from sample to sample) and then checks its correlation with the parent peak. This is done to ensure that the isotope El-MAVEN is reporting is actually an isotope and not some other metabolite with the same m/z as an isotope.

It does miss a lot of isotopic peaks this way, though, you're right. Making isotope detection slightly less dependent on the parent peak will go a long way. Such improvements in isotope detection are a part of the roadmap we shared with you in November. If all goes according to plan, the April/May release should have the required changes. We're working on finding a solution to this problem and will reach out for your input and feedback on the same

**Comment by shubhra-agrawal** [comment]
@paopaocui has brought up this concern as well in #1046. Closing it as a duplicate.

@surbhipoddar Lets move this up on our priority list.
